include/xtl/xbasic_fixed_string.hpp

```diff
@@ -67,7 +68,56 @@ namespace xtl
         template <int selector>
         struct select_storage;
 
+        template <typename T>
+        struct fixed_small_string_storage_impl;
+
         template <class T, std::size_t N>
+        struct fixed_small_string_storage_impl<T[N]>
+        {
+            static_assert(N <= (1u << (8 * sizeof(T))), "small string");
+
+            fixed_small_string_storage_impl() {
+              set_size(0);
+            }
+
+            fixed_small_string_storage_impl(T ptr[N], std::size_t size)
+                : m_buffer(ptr)
+            {
+              m_buffer[N - 1] = N - size;
+            }
+
+            auto& buffer()
```

Comment:
Nitpicking: I think you can return T& instead of auto

Reply:
I don't think so, `m_buffer` has type `T[N]`

Reply:
My bad, I read too fast